### ISSUE TITLE
mbed crashes & reboots periodically

### ISSUE THREAD

**Issue by justbuchanan** [RoboJackets/robocup-software#686]
It might be that it's running out of memory, but we're not quite sure.

cc @jjones646

**Comment by jjones646** [comment]
Almost certainly a memory issue. I've been running everything through soccer for 15+ min now using ca04d6f, and it's been doing great.

**Comment by jjones646** [comment]
This line in Console.cpp definitely isn't helping the issue...

```
// reserve space for 5 lines in rx buffer
_rxBuffer.reserve(400);
```

This seems to be heap allocation.

**Comment by jjones646** [comment]
Another thing to note is that `_rxBuffer` is a `std::string` object. And it looks like under allocation is fine since all additions use `_rxBuffer.push_back`. So a reservation of 10 or 15 is more reasonable.

**Comment by justbuchanan** [comment]
This file in the mbed repo has some info on stack sizes for the main thread and timer thread: https://github.com/mbedmicro/mbed/blob/master/libraries/doc/rtos.txt.

We would have to fork the mbed repo, but it may be worth looking into reducing both of those threads' sizes - main is definitely a lot bigger than it needs to be since most of our stuff happens in other threads.

**Comment by justbuchanan** [comment]
Mostly fixed in #707
